Tarantool fails to build on Linux ARM64 with CMAKE_BUILD_TYPE=Debug

**Bug description**

Tarantool Debug configuration fails to build on our Linux ARM64 hosts due to some warnings in `http_parser.c`, since `-Werror` is enabled unconditionally for Debug. If one disables backtraces (see #6142) and set release build type on configuration phase, Tarantool is successfully built.

* OS:
```
$ cat /etc/os-release 
NAME="Ubuntu"
VERSION="18.04.4 LTS (Bionic Beaver)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 18.04.4 LTS"
VERSION_ID="18.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=bionic
UBUNTU_CODENAME=bionic
$ uname -rs
Linux 4.9.213-67
```
* Architecture: 
```
$ uname -m
aarch64
```

**Steps to reproduce**

```
$ git clone https://github.com/tarantool/tarantool.git tbuild
<...>
$ cd !$
$ cmake . -DENABLE_BACKTRACE=OFF # see more info in #6142 
<...>
$ make
<...>
Scanning dependencies of target http_parser
[ 26%] Building C object src/lib/http_parser/CMakeFiles/http_parser.dir/http_parser.c.o
/home/imun/tbuild/src/lib/http_parser/http_parser.c: In function ‘http_parse_header_line’:
/home/imun/tbuild/src/lib/http_parser/http_parser.c:272:12: error: comparison is always false due to limited range of data type [-Werror=type-limits]
     if (ch < 0) {
            ^
/home/imun/tbuild/src/lib/http_parser/http_parser.c:309:11: error: comparison is always false due to limited range of data type [-Werror=type-limits]
    if (ch < 0) {
           ^
/home/imun/tbuild/src/lib/http_parser/http_parser.c: At top level:
cc1: error: unrecognized command line option ‘-Wno-cast-function-type’ [-Werror]
cc1: error: unrecognized command line option ‘-Wno-gnu-alignof-expression’ [-Werror]
cc1: all warnings being treated as errors
src/lib/http_parser/CMakeFiles/http_parser.dir/build.make:62: recipe for target 'src/lib/http_parser/CMakeFiles/http_parser.dir/http_parser.c.o' failed
make[2]: *** [src/lib/http_parser/CMakeFiles/http_parser.dir/http_parser.c.o] Error 1
CMakeFiles/Makefile2:5084: recipe for target 'src/lib/http_parser/CMakeFiles/http_parser.dir/all' failed
make[1]: *** [src/lib/http_parser/CMakeFiles/http_parser.dir/all] Error 2
Makefile:151: recipe for target 'all' failed
make: *** [all] Error 2
```

**Actual behavior**

```
$ file src/tarantool
src/tarantool: cannot open `src/tarantool' (No such file or directory)
```

**Expected behavior**

```
$ file src/tarantool
src/tarantool: ELF 64-bit LSB shared object, ARM aarch64, version 1 (SYSV), dynamically linked, interpreter /lib/ld-, for GNU/Linux 3.7.0, BuildID[sha1]=ec4613f909d87f4deec9c9314b2d967b100f5599, with debug_info, not stripped
```

I'd like to underline that this issue also raises in fedora builds.
https://github.com/tarantool/tarantool/runs/3180357806?check_suite_focus=true#step:5:1271
https://github.com/tarantool/tarantool/runs/3180357642?check_suite_focus=true#step:5:1341


For `Fedora 33 - aarch64`: 

`/build/usr/src/debug/tarantool-2.9.0.167/src/lib/http_parser/http_parser.c: In function ‘http_parse_header_line’:
/build/usr/src/debug/tarantool-2.9.0.167/src/lib/http_parser/http_parser.c:272:12: error: comparison is always false due to limited range of data type [-Werror=type-limits]
  272 |     if (ch < 0) {
      |            ^
/build/usr/src/debug/tarantool-2.9.0.167/src/lib/http_parser/http_parser.c:309:11: error: comparison is always false due to limited range of data type [-Werror=type-limits]
  309 |    if (ch < 0) {
      |           ^`

--- a/changelogs/unreleased/gh-6143-arm64-debug-build
+++ b/changelogs/unreleased/gh-6143-arm64-debug-build
@@ -1,0 +1,3 @@
+## bugfix/build
+
+ * Fix build errors on arm64 with `CMAKE_BUILD_TYPE=Debug`.

--- a/src/lib/http_parser/http_parser.c
+++ b/src/lib/http_parser/http_parser.c
@@ -221,7 +221,8 @@ int
 http_parse_header_line(struct http_parser *prsr, char **bufp,
 		       const char *end_buf, int max_hname_len)
 {
-	char c, ch;
+	char c;
+	unsigned char ch;
 	char *p = *bufp;
 	char *header_name_start = p;
 	prsr->hdr_name_idx = 0;
@@ -268,10 +269,6 @@ http_parse_header_line(struct http_parser *prsr, char **bufp,
 				goto header_done;
 			default:
 				state = sw_name;
-
-				if (ch < 0) {
-					return HTTP_PARSE_INVALID;
-				}
 				c = lowcase[ch];
 				if (c != 0) {
 					prsr->hdr_name[0] = c;
@@ -306,9 +303,6 @@ http_parse_header_line(struct http_parser *prsr, char **bufp,
 			break;
 		/* http_header name */
 		case sw_name:
-			if (ch < 0) {
-				return HTTP_PARSE_INVALID;
-			}
 			c = lowcase[ch];
 			if (c != 0) {
 				if (prsr->hdr_name_idx < max_hname_len) {

